safe/gui/gui_utilities.py

```diff
@@ -0,0 +1,53 @@
+# coding=utf-8
+
+"""GUI utilities for the dock and the multi Exposure Tool."""
+
+from PyQt4.QtCore import Qt
+
+from qgis.core import QgsMapLayerRegistry
+
+
+def layer_from_combo(combo):
+    """Get the QgsMapLayer currently selected in a combo.
+
+    Obtain QgsMapLayer id from the userrole of the QtCombo and return it as a
+    QgsMapLayer.
+
+    :returns: The currently selected map layer a combo.
+    :rtype: QgsMapLayer
+    """
+    index = combo.currentIndex()
+    if index < 0:
+        return None
+
+    layer_id = combo.itemData(index, Qt.UserRole)
+    layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
+    return layer
+
+
+def add_ordered_combo_item(combo, text, data=None):
+    """Add a combo item ensuring that all items are listed alphabetically.
+
+    Although QComboBox allows you to set an InsertAlphabetically enum
+    this only has effect when a user interactively adds combo items to
+    an editable combo. This we have this little function to ensure that
+    combos are always sorted alphabetically.
+
+    :param combo: Combo box receiving the new item.
+    :type combo: QComboBox
+
+    :param text: Display text for the combo.
+    :type text: str
+
+    :param data: Optional UserRole data to be associated with the item.
+    :type data: QVariant, str
+    """
+    size = combo.count()
+    for combo_index in range(0, size):
+        item_text = combo.itemText(combo_index)
+        # see if text alphabetically precedes item_text
+        if cmp(text.lower(), item_text.lower()) < 0:
```

Comment:
__[flake8]__
* [F821] undefined name 'cmp'

<!-- Comment by [SideCI](https://sideci.com) -->

Reply:
This was done before in the previous function. It will be removed when we will switch to Python 3